packages/main/src/controllers/credentials/CredentialTypeController.ts

```diff
@@ -57,11 +59,23 @@ export class CredentialTypesController {
 
         const schema = schemaResult.schema
 
+        const revocationDefinitionRepository = agent.dependencyManager.resolve(
+          AnonCredsRevocationRegistryDefinitionRepository,
+        )
+        const revocationRegistries = await revocationDefinitionRepository.findAllByCredentialDefinitionId(
+          agent.context,
+          record.credentialDefinitionId,
+        )
+        const revocationRegistryDefinitionIds = revocationRegistries.map(
+          item => item.revocationRegistryDefinitionId,
+        )
+
         return {
           id: record.credentialDefinitionId,
           name: (record.getTag('name') as string) ?? schema?.name,
           version: (record.getTag('version') as string) ?? schema?.version,
           attributes: schema?.attrNames || [],
+          revocationIds: revocationRegistryDefinitionIds,
```

Comment:
I still don't see the point of including the ids of revocation registry ids associated to the credential definition. Is it only for convenience on testing with the chatbot demo? If that's the case, I don't think it is the right approach.

You can also create a GET endpoint (like this one) that returns all revocation registry definitions and even add a parameter to filter them by credential definition id. Then, in the chatbot demo you do another request to Service Agent to query for the revocation registries and use the first one related to this credential type.

Reply:
fix: updated with separation